### PR TITLE
Refactor PersonBuilder constructor to use TypicalPersons as argument #855

### LINKED ISSUE #855: SystemTests: Refactor PersonBuilder to use TypicalPersons as arguments
In many cases, to create a person for testing, the constants from `CommandTestUtil` is used with `PersonBuilder` like so: 
```java
        toAdd = new PersonBuilder().withName(VALID_NAME_AMY).withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_AMY)
                .withAddress(VALID_ADDRESS_AMY).withTags(VALID_TAG_FRIEND).build();
```

This way of creating persons is long-winded and is restricted to only two Persons (Amy and Bob). 

Let's refactor these usages to a more concise form like so:
```java
toAdd = new PersonBuilder(AMY).withPhone(VALID_PHONE_BOB).build()
```

### PR DESCRIPTION
**Pull request by LiXiaoooowei**
Fixes #855

### CODE REVIEW

**Comment by LiXiaoooowei** [comment]
Ready for review

**Comment by yamgent** [comment]
Hi @LiXiaoooowei, thanks for your PR!

Good job on your first PR so far. However, some changes will be needed in order for us to continue with this PR:

### First commit: Refactor code to initiate PersonBuilder by accepting Person instances…
1. Commit message subject is too long. It should not be longer than 72 characters, otherwise GitHub will break the message and we will see … 
    * See the [Commit message convention](https://github.com/oss-generic/process/blob/master/docs/FormatsAndConventions.adoc#commit-message) for more details regarding commit message subject.
    * You will have to be creative here since the issue title is quite long, but it can certainly be summarized without using a lot of words.
2. Commit message body is empty. The purpose of the message body is to explain the rationale for the changes made, so that future developers will be able to look back at your commits to get a better understanding of what is going on.
    * See the [Commit message convention](https://github.com/oss-generic/process/blob/master/docs/FormatsAndConventions.adoc#commit-message) for more details on how to write the commit message body.
    * The rationale of the changes can be found in the original post of #855.
    * Do take note that the body should be wrapped.

### Second commit: Remove unused imports and fix style issue
3. Squash this commit with the first commit. We should squash this commit because it is relevant to the first commit. The practices of se-edu differs from common practices of other OSS projects. We practice squashing in order to keep the history clean, and each commit _by itself_ should have their test passing.
    * See the [Commit organization convention](https://github.com/oss-generic/process/blob/master/docs/FormatsAndConventions.adoc#commit-organization) for more details about the need to squash.
    * There should be a few tutorials online on what is squashing and how to do the squashing. For an authoritative source, see [this tutorial](https://comp.yamgent.com/2018/06/30/squashing-git-cleaning-up.html), which is definitely not written by a rather familiar person 🤔.
    
### Submitting code for reviews
4. We use a commit summary tool, known as `CanIHasReview`, to submit code for reviews, instead of posting a "Ready for Review" comment. 
    * See the [Workflow convention](https://github.com/oss-generic/process/blob/master/docs/QuestionsIssuesPrs.adoc#326-request-a-review) to see how you can submit your PR for further review.

### Others
5. Branch name is too long. For long issues, the [convention is to use 3-5 keywords](https://github.com/oss-generic/process/blob/master/docs/FormatsAndConventions.adoc#branch). E.g.: `889-personbuilder-use-typicalpersons` (still a bit long than desired, but it fits the general guidelines of 3-5 keywords).
    * Note: I don't think we can rename a branch without having to re-open the PR, so for now, we can live with this branch name. Just take note of this for subsequent PRs.
6. Btw, don't need to start the branch name with `#`, just the issue number will do (i.e. `889-personbuilder-use-typicalpersons`, not `#889-personbuilder-use-typicalpersons`).

Do let me know if you have any more questions. Thanks!

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 1]
```diff
@@ -71,18 +68,16 @@ public void parse_allFieldsPresent_success() {
                 + ADDRESS_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedPerson));
 
         // multiple tags - all accepted
-        Person expectedPersonMultipleTags = new PersonBuilder().withName(VALID_NAME_BOB).withPhone(VALID_PHONE_BOB)
-                .withEmail(VALID_EMAIL_BOB).withAddress(VALID_ADDRESS_BOB)
-                .withTags(VALID_TAG_FRIEND, VALID_TAG_HUSBAND).build();
+        Person expectedPersonMultipleTags = new PersonBuilder(BOB).withTags(VALID_TAG_FRIEND, VALID_TAG_HUSBAND)
+                .build();
```
Comment (yamgent) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: I think this is just equivalent to `BOB`. Just assign `BOB` directly?

**Comment by yamgent** [comment]
Good attempt in writing the commit message. :+1:

Commit message paragraphs need to follow a certain style and structure. Do take a look at the ['stylistic recommendations' portion of the Commit message convention](https://github.com/oss-generic/process/blob/master/docs/FormatsAndConventions.adoc#commit-message) to find out more. Some examples of writing commit messages are also given, you can refer to those to further understand how a commit message is structured.

For your convenience, I have also given my suggested changes on how you can improve the commit message. Do take note that my English is not necessary the best, so if there's any errors, feel free to change as necessary:

----

> ```
> Initiate PersonBuilder with TypicalPerson instance when necessary
> ```

* I just realize that the original issue's title doesn't really make sense, so I have taken the liberty to come up with a better subject title that suits the context of the commit. 
* The words "when necessary" is not needed, it is implied.

Suggested change for the title:

```
Tests: Shorten calls to PersonBuilder
```

----

> ```
> To create a Person for testing, constants from CommandTestUtil is used
> with default PersonBuilder constructor(eg. toAdd = new PersonBUilder()
> .withName(VALID_NAME_AMY).withPhone(VALID_PHONE_AMY).withAddress(VALID
> _ADDRESS_BOB).build()).
> ```

* There is a typo in the example (`new PersonBUilder()`).
* The examples neglected the tags field (`withTags()`), so they are actually showing the _wrong_ way of how to refactor the code.
* While the body text must be wrapped at 72 characters, one type of text can violate this rule - stuff that gets very ugly when wrapped, such as codes and shell commands.

Suggested change for this paragraph (this is the paragraph for `{current situation}`):

```
When creating a Person for testing, some tests specify almost all of
the fields of an existing TypicalPerson, with only a slight variation
in one or two fields. For example, the code below builds on top of
TypicalPersons#AMY with only changes to address and tags:

    toAdd = new PersonBuilder().withName(VALID_NAME_AMY).withPhone(VALID_PHONE_AMY)
            .withEmail(VALID_EMAIL_AMY).withAddress(VALID_ADDRESS_BOB).withTags().build();
```

----

> ```
> This is long-winded and restricted to only AMY and BOB.
>
> We can use a more concise and extensible form(eg. toAdd = new
> PersonBuilder(AMY).withAddress(VALID_ADDRESS_BOB).
> ```

* Don't think the mentioning of AMY and BOB outside the code example would be necessary. This will be apparent in the commit diff.
* These two paragraphs should be merged together because they both belong to `{why it needs to change}`.

Suggested change for this paragraph (this is the paragraph for `{why it needs to change}`):

```
This is long-winded and unnecessary, as PersonBuilder allows us to start
from a ready-made Person, and modify as necessary. For example, the code
above can be refactored to:

    toAdd = new PersonBuilder(AMY).withAddress(VALID_ADDRESS_BOB).withTags().build();    
```

----

* The paragraph for `{what is being done about it}` is missing.

Suggested addition to the commit message (this is the paragraph for `{what is being done about it}`):

```
Let's refactor the calls to PersonBuilder to make them more concise.
```

----

With everything combined, it should be like this:

```
Tests: Shorten calls to PersonBuilder

When creating a Person for testing, some tests specify almost all of
the fields of an existing TypicalPerson, with only a slight variation
in one or two fields. For example, the code below builds on top of
TypicalPersons#AMY with only changes to address and tags:

    toAdd = new PersonBuilder().withName(VALID_NAME_AMY).withPhone(VALID_PHONE_AMY)
            .withEmail(VALID_EMAIL_AMY).withAddress(VALID_ADDRESS_BOB).withTags().build();

This is long-winded and unnecessary, as PersonBuilder allows us to start
from a ready-made Person, and modify as necessary. For example, the code
above can be refactored to:

    toAdd = new PersonBuilder(AMY).withAddress(VALID_ADDRESS_BOB).withTags().build(); 

Let's refactor the calls to PersonBuilder to make them more concise.    
```

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java (hunk 1 above)
Comment (LiXiaoooowei) [reply src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: The test is to examine parser behaviour in the case of multiple tags, I am not sure if removing `.withTags(VALID_TAG_FRIEND, VALID_TAG_HUSBAND)` makes the purpose of this line of code unclear to reader?

Comment (yamgent) [reply src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: You do have a point. Ok, in that case, don't need to change this line.

**Comment by yamgent** [comment]
@LiXiaoooowei nice work. You can progress to harder issues for your next PR.

**Comment by LiXiaoooowei** [comment]
@yamgent Noted. Thank you so much!

**Pull request merged**